src/executor/nested_loop_join_executor.cpp

```diff
@@ -74,26 +76,23 @@ bool NestedLoopJoinExecutor::DInit() {
  * to join the left and right result.
  *
  */
+
+// For this version, the work flow is that we first lookup the left table, and
+// use the result to lookup right table. If left table is done that means right
+// table is also done. So we only keep the left_child_done_ as the sign.
 bool NestedLoopJoinExecutor::DExecute() {
```

Comment:
Does this method still work with non-index nested loop join (sequential scan nested loop join) after your modification? Did you test that?